yt/yt/library/decimal/decimal.cpp

```diff
@@ -12,11 +12,129 @@ namespace NYT::NDecimal {
 
 ////////////////////////////////////////////////////////////////////////////////
 
+// We use the same type used for binary representation of 256 bit decimals for
+// implementing the necessary arithmetic operations.
+using i256 = TDecimal::TValue256;
+
+// We chose to only implement a small subset of operations and functions necessary
+// for converting 256 bit decimals to and from text.
+// Rationale: there do not seem to be any good implementations of 256-bit arithmetic
+// that we are comfortable depending on (for now). There are some big number
+// implementations around openssl, but these types are intended for arbitrarily
+// large integers and use dynamic allocations. It also does not make sense to commit
+// to implementing full-fledged 256-bit arithmetics when we actually use a minority
+// of operations.
+// When we have a good enough int256 either in library/util or in contrib, we can
+// switch to it easily.
+
+////////////////////////////////////////////////////////////////////////////////
+
+constexpr i256 operator-(i256 value) noexcept
+{
+    // Invert.
+    for (int partIndex = 0; partIndex < 8; ++partIndex) {
```

Comment:
Let's use part.size() or range-based fors here? It is known in a compile time but makes less space for an error